app/controller_services/shopping_list_items_controller/create_service.rb

```diff
@@ -24,16 +24,18 @@ def perform
       item = ShoppingListItem.combine_or_new(params.merge(list_id:))
 
       ActiveRecord::Base.transaction do
+        lists_changed = lists_to_be_changed
```

Comment:
Unfortunately, this method needs to be called at this point instead of just adding it on lines 34 and 38 like I had hoped. The reason is because, by line 34/38, if the unit weight is being changed, it has already been updated on the aggregate list item and we can no longer tell if it was changed.